lib/internal/fs/streams.js

```diff
@@ -131,6 +131,9 @@ ReadStream.prototype.open = function() {
       return;
     }
 
+    if (this.destroyed)
+      return;
```

Comment:
This creates a file descriptor leak, as the `_destroy` logic relies on the `'open'` even. You'll need to move this bit https://github.com/nodejs/node/blob/0e06b350b6d7c80875321531593efc6f273620e5/lib/internal/fs/streams.js#L209 in here and tear down the file descriptor.

The same applies to `WriteStream`.

Reply:
So should the `close` event still be emitted?

Reply:
@mcollina

Reply:
I mean that you should close the file descriptor in the `if` block.